### ISSUE TITLE
[feature request] Batch size scheduler

### ISSUE THREAD

**Issue by mariosfourn** [pytorch/pytorch#10470]
It would be useful to have had a batch size scheduler similar to learning rate.  Recent research suggest that this is a better and more efficient optimisation practice.

https://arxiv.org/abs/1711.00489

**Comment by soumith** [comment]
The paper is about increasing the batch size by increasing the effective number of machines / doing more parallelization.

Decreasing the learning rate, or increasing the batch size is equivalent (according to the paper).

I'm not sure how easy it is from an API design perspective to add a batch size scheduler, as it interacts with DataLoader.

It's trivial to do it in user code, using a standard learning rate scheduler (that we provide), and then just doing `batch_size = initial_batch_size * (initial_learning_rate / current_learning_rate)`.

Because it's not a trivial API design, and it's trivial to handle it in user code, I'm closing the issue.

**Comment by drevicko** [comment]
@soumith That's not entirely accurate. They cite Goyal et.al. that keeping the ratio of batch size to learning rate constant results in similar generalisability. This means you can run with larger batch sizes and larger learning rates whilst maintaining test set performance, which trains faster if you can do batches in parallel.  

They propose a schedule on this ratio, which can be achieved by constant batch size (standard lr scheduling) or by varying batch size with constant lr (what @mariosfourn is suggesting) or something between (which they actually did in their experiments). 

I admit that doing this is trivial since user code has full control over batch size, hence no great need for a scheduler class.
